test/ex_doc/formatter/html_test.exs

```diff
@@ -592,6 +592,56 @@ defmodule ExDoc.Formatter.HTMLTest do
              ] = Jason.decode!(content)["extras"]
     end
 
+    test "custom search data is added to the sidebar and search nodes",
+         %{tmp_dir: tmp_dir} = context do
+      generate_docs(
+        doc_config(context,
+          source_beam: "unknown",
+          extras: [
+            {"test/fixtures/README.md",
+             search_data: [
+               %{
+                 anchor: "",
+                 title: "top of the doc",
+                 type: "custom",
+                 body: """
+                 In this doc we...
+                 """
+               },
+               %{
+                 anchor: "heading-without-content",
+                 title: "custom-text",
+                 type: "custom",
+                 body: """
+                 Some longer text!
+
+                 Here it is :)
+                 """
+               }
+             ]}
+          ]
+        )
+      )
+
+      "sidebarNodes=" <> content = read_wildcard!(tmp_dir <> "/html/dist/sidebar_items-*.js")
```

Comment:
We need to test search_data-*.js too, sorry for all the back and forth!

Reply:
Not a problem :)  It's not right until it's right!